src/pages/NewsList/NewsList.tsx

```diff
@@ -44,68 +34,79 @@ export default function NewsList() {
 
 	useEffect(() => {
 		const fetchNewsData = async () => {
-			console.log('Fetch data: trying to fetch');
+			console.log('Fetch data: fetch started');
 			try {
 				if (!apiKey) {
 					throw new Error('API key not available');
 				}
 
-				const response = await fetch(`https://www.alphavantage.co/query?function=NEWS_SENTIMENT&tickers=IBM&limit=10000&apikey=${apiKey}`);
+				const response = await fetch(apiUrl);
 
 				if (!response.ok) {
 					throw new Error('Network response was not ok');
 				}
 
 				const data = await response.json();
 
-				// update localStorage only if data is not empty, to avoid undefined, empty object
-				if (data.length > 0) {
-					localStorage.setItem('cachedNewsData', JSON.stringify(data));
+				// due to api daily limit, store data into localStorage only when data from server is not empty to avoid undefined object
+				if (data && data.feed && data.feed.length > 0) {
+					localStorage.setItem('cachedNewsData', JSON.stringify(data.feed));
+					console.log('Fetch data: data also stored in localStorage');
+					handleFetchSuccess(data.feed);
+				} else {
+					handleFetchError(new Error('No data fetched from API'));
 				}
-
 				handleFetchSuccess(data.feed);
 			} catch (error) {
 				handleFetchError(error);
 			}
 		};
-		fetchNewsData();
-		console.log('Fetch data: data fetched from server via api call');
-	}, [apiKey]);
+
+		fetchNewsData().then(r => {
+		});
+		console.log('Fetch data: data fetched');
+	}, []);
+
+	const data = newsData && newsData.length > 0 ? newsData : (cachedData ? JSON.parse(cachedData) : []);
```

Comment:
eslint would hate that :D nested ternary is unreadable and usually prohibited in the default config.

Reply:
strange, eslint is silent. using our config, extended for js, react, ts. rewritten into if else.